src/main/java/io/anserini/fusion/ScoredDocsFuser.java

```diff
@@ -0,0 +1,271 @@
+
+import io.anserini.search.ScoredDocs;
+
+//replace topic wtih const
+public class ScoredDocsFuser {
+  public static final String TOPIC = "TOPIC";
+
+  /**
+   * Reads a TREC run file and returns a ScoredDocs containing the data.
+   * 
+   * @param filepath Path to the TREC run file.
+   * @throws IOException If the file cannot be read.
+   * @return A ScoredDocs object containing the data from the TREC run file.
+   */
+  public static ScoredDocs readRun(Path filepath, boolean reSort) throws IOException {
+    ScoredDocs scoredDocs = new ScoredDocs();
+    try (BufferedReader br = new BufferedReader(new FileReader(filepath.toFile()))) {
+      List<Document> lucene_documents = new ArrayList<>(); // topic
+      List<String> docids = new ArrayList<>(); // docid
+      List<Float> scores = new ArrayList<>(); // score
+      List<Integer> rank = new ArrayList<>(); // rank
+
+      String line;
+      while ((line = br.readLine()) != null) {
+        String[] data = line.split("\\s+");
+  
+        // Populate the lists with the parsed topic and docid
+        Document doc = new Document();
+        doc.add(new StoredField(TOPIC, data[0]));
+        lucene_documents.add(doc);
+        docids.add(data[2]);
+
+        // Parse RANK as integer
+        int rankInt = Integer.parseInt(data[3]);
+        rank.add(rankInt);
+
+        // Parse SCORE as float
+        float scoreFloat = Float.parseFloat(data[4]);
+        scores.add(scoreFloat);
+      }
+
+      scoredDocs.lucene_documents = lucene_documents.toArray(new Document[0]);
+      scoredDocs.docids = docids.toArray(new String[0]);
+      scoredDocs.scores = ArrayUtils.toPrimitive(scores.toArray(new Float[scores.size()]), Float.NaN);
+      scoredDocs.lucene_docids = ArrayUtils.toPrimitive(rank.toArray(new Integer[0]));
+    }
+  
+    if (reSort) {
+      ScoredDocsFuser.sortScoredDocs(scoredDocs);
+    }
+
+    return scoredDocs;
+  }
+
+  /**
+   * Rescored given ScoredDocs using the specified method.
+   *
+   * @param method  Rescore method to be applied (e.g., RRF, SCALE, NORMALIZE).
+   * @param rrfK    Parameter k needed for reciprocal rank fusion.
+   * @param scale   Scaling factor needed for rescoring by scaling.
+   * @param scoredDocs ScoredDocs object to be rescored.
+   * @throws UnsupportedOperationException If an unsupported rescore method is provided.
+   */
+  public static void rescore(RescoreMethod method, int rrfK, double scale, ScoredDocs scoredDocs) {
+    switch (method) {
+      case RRF -> ScoredDocsFuser.rescoreRRF(rrfK, scoredDocs);
+      case SCALE -> ScoredDocsFuser.rescoreScale(scale, scoredDocs);
+      case NORMALIZE -> ScoredDocsFuser.normalizeScores(scoredDocs);
+      default -> throw new UnsupportedOperationException("Unknown rescore method: " + method);
+    }
+  }
+
+  private static void rescoreRRF(int rrfK, ScoredDocs scoredDocs) {
+    int length = scoredDocs.lucene_documents.length;
+    for(int i = 0; i < length; i++){
```

Comment:
nit, spaces ^
```suggestion
    for (int i = 0; i < length; i++) {
```

change for all instances please.